### PR TITLE
Cmake Installation Functionality

### PR DESCRIPTION
**Pull request by FelixBenning**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ enable_testing()
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
   set(gtest_force_shared_crt ON CACHE BOOL "Make gtest compile the C standard libraries as DLLs, because MSVE requires that." FORCE)
 endif()
-add_subdirectory(googletest)
+add_subdirectory(googletest EXCLUDE_FROM_ALL)
```
Comment (eike-fokken) [inline CMakeLists.txt:48]: I see that this works and seems to also build tests but I don't understand it.
How comes, googletest is actually build when I run cmake --build ...?

**Comment by eike-fokken** [comment]
If the gui is tricky, let us forego it for now.
In the medium-to-long term it will be replaced by just calling grazer from python (or possibly matlab, but thats more difficult).
Is there an easy way for me to just merge the first commit and are there things that still need to be done before I merge it?

File: CMakeLists.txt (hunk 1 above)
Comment (FelixBenning) [reply CMakeLists.txt:48]: This only excludes that directory from installs https://stackoverflow.com/a/64900982/6662425

**Comment by eike-fokken** [comment]
I'm closing this for now and will merge the first commit in #224

**Pull request closed without merging**